### PR TITLE
[DOCS-858] Adding log processing rule examples for docker and K8s

### PR DESCRIPTION
**Pull request by l0k0ms**
### What does this PR do?

* Adds example for docker and Kubernetes for advanced log processing rules
* Creates a dedicated FAQ with examples for commonly used one.

### Motivation

Better reflexion of the log collections features across platforms.

### Preview link

* [Docker log collection](https://docs-staging.datadoghq.com/gus/log-collection-docker-kuberentes/agent/docker/log/?tab=containerinstallation#advanced-log-collection)
* [Kubernetes log collection](https://docs-staging.datadoghq.com/gus/log-collection-docker-kuberentes/agent/kubernetes/log/?tab=daemonset#advanced-log-collection)
* [New advanced log collection page](https://docs-staging.datadoghq.com/gus/log-collection-docker-kuberentes/agent/logs/advanced_log_collection?tab=configurationfile)
* [New commonly used FAQ](https://docs-staging.datadoghq.com/gus/log-collection-docker-kuberentes/agent/faq/commonly-used-log-processing-rules/)

### CODE REVIEW

File: content/en/agent/faq/commonly-used-log-processing-rules.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+---
+title: Commonly Used Log Processing Rules
```
Comment (PierreVanRoy) [inline content/en/agent/faq/commonly-used-log-processing-rules.md:2]: I'm not sure that this is a good title for this page as it doesn't really cover "common processing rules" but only different setting to deal with sensitive data. Could this be a guide  called "How to deal with sensitive data" more than a FAQ ? 
I would rather organize this page in 3 sections:
- mask sensitive data (listing all the different example  of 'mask_sequences' in sub sections)
- exclude logs including sensitive data (blacklist)
- include only safe logs (whitelist) (section to add referring to include_at_match )

File: content/en/agent/logs/advanced_log_collection.md [hunk 2]
```diff
@@ -58,13 +60,78 @@ logs:
...
+      containers:
+        - name: cardpayment
+          image: cardpayment:latest
+```
+
+**Note**: Escape regex characters in your patterns when using pod annotations. For example, `\d` becomes `\\d`, `\w` becomes `\\w`, etc.
+
+{{% /tab %}}
+{{< /tabs >}}
+
```
Comment (PierreVanRoy) [inline content/en/agent/logs/advanced_log_collection.md:123]: Add a link to the FAQ/guide to So the the user can have more example of classique scrubbing rules.

File: content/en/agent/faq/commonly-used-log-processing-rules.md (hunk 1 above)
Comment (l0k0ms) [reply content/en/agent/faq/commonly-used-log-processing-rules.md:2]: for now there are only example on how to deal with sensitive data, but soon there will be more rules documented here :) but i agree on the organization proposal

File: content/en/agent/logs/advanced_log_collection.md (hunk 2 above)
Comment (l0k0ms) [reply content/en/agent/logs/advanced_log_collection.md:123]: link is in its own section already on the page :)

File: content/en/agent/docker/log.md [hunk 3]
```diff
@@ -202,17 +202,25 @@ See the [multi-line processing rule documentation][1] to get more pattern exampl
 - Use the environment variable to collect logs from all containers and then override the default `source` and `service` values.
 - Add processing rules for the wanted subset of containers.
 
+## Advanced log collection
+
+Use the autodiscovery log label to apply an advanced log collection processing logic like:
```
Comment (cswatt) [inline content/en/agent/docker/log.md:207]:
```suggestion
Use Autodiscovery log labels to apply advanced log collection processing logic, for example:
```

File: content/en/agent/kubernetes/log.md [hunk 4]
```diff
@@ -369,9 +369,18 @@ Unlike auto-conf files, **key-value stores may use the short OR long image name
 {{% /tab %}}
 {{< /tabs >}}
 
+
+## Advanced log collection
+
+Use the autodiscovery log annotation to apply an advanced log collection processing logic like:
```
Comment (cswatt) [inline content/en/agent/kubernetes/log.md:375]:
```suggestion
Use Autodiscovery log labels to apply advanced log collection processing logic, for example:
```

File: content/en/agent/logs/_index.md [hunk 5]
```diff
@@ -24,16 +24,15 @@ Log collection requires the Datadog Agent v6.0+. Older versions of the Agent do
 
 ## Activate log collection
 
-Collecting logs is **disabled** by default in the Datadog Agent.
+Collecting logs is **disabled** by default in the Datadog Agent. To lean how to enable it with the host Agent see the instructions below, if you are running the Agent in a kuberentes or docker environment, see the dedicated. [Kubernetes Log Collection][2] and [Docker Log Collection][3] documentations.
```
Comment (cswatt) [inline content/en/agent/logs/_index.md:27]:
```suggestion
Collecting logs is **disabled** by default in the Datadog Agent. To lean how to enable it with the host Agent, see the instructions below. If you are running the Agent in a Kuberentes or Docker environment, see the dedicated [Kubernetes Log Collection][2] or [Docker Log Collection][3] documentation.
```

File: content/en/agent/logs/_index.md [hunk 6]
```diff
@@ -24,16 +24,15 @@ Log collection requires the Datadog Agent v6.0+. Older versions of the Agent do
...
-To enable log collection, update the Agent's [main configuration file][10] (`datadog.yaml`) with:
+To enable log collection with an Agent running on your host, update the Agent's [main configuration file][4] (`datadog.yaml`) with:
 
 ```yaml
 logs_enabled: true
 ```
 
-Starting with Agent v6.19+/v7.19+, HTTPS transport is the default transport used.
-For more details on how enforce HTTPS/TCP transport, refer to the [Agent transport documentation][11].
+Starting with Agent v6.19+/v7.19+, HTTPS transport is the default transport used. For more details on how enforce HTTPS/TCP transport, refer to the [Agent transport documentation][5].
```
Comment (cswatt) [inline content/en/agent/logs/_index.md:35]:
```suggestion
Starting with Agent v6.19+/v7.19+, HTTPS transport is the default transport used. For more details on how to enforce HTTPS/TCP transport, refer to the [Agent transport documentation][5].
```

File: content/en/agent/logs/_index.md [hunk 7]
```diff
@@ -43,23 +42,21 @@ After activating log collection, the Agent is ready to forward logs to Datadog.
 
 ## Enabling log collection from integrations
 
-To collect logs for a given integration, uncomment the logs section in that integration's `conf.yaml` file and configure it for your environment.
+To collect logs for a given integration, uncomment the logs section in that integration's `conf.yaml` file and configure it for your environment. If you are running the Agent in a kubernetes or docker environment, see the dedicated. [Kubernetes Log Collection][6] and [Docker Log Collection][7] documentations.
```
Comment (cswatt) [inline content/en/agent/logs/_index.md:45]:
```suggestion
To collect logs for a given integration, uncomment the logs section in that integration's `conf.yaml` file and configure it for your environment. If you are running the Agent in a Kubernetes or Docker environment, see the dedicated [Kubernetes Log Collection][6] or [Docker Log Collection][7] documentation.
```

File: content/en/agent/logs/advanced_log_collection.md [hunk 8]
```diff
@@ -35,14 +35,16 @@ To apply a processing rule to all logs collected by a Datadog Agent, see the [Gl
...
-{{< tabs >}}
-{{% tab "exclude_at_match" %}}
+### exclude_at_match
 
 | Parameter          | Description                                                                                        |
 |--------------------|----------------------------------------------------------------------------------------------------|
 | `exclude_at_match` | If the specified pattern is contained in the message, the log is excluded and not sent to Datadog. |
 
-For example, filtering out logs that contain a Datadog email address:
+For example, **filtering OUT** logs that contain a Datadog email address, use the following `log_processing_rules`:
```
Comment (cswatt) [inline content/en/agent/logs/advanced_log_collection.md:44]:
```suggestion
For example, to **filter OUT** logs that contain a Datadog email address, use the following `log_processing_rules`:
```

File: content/en/agent/logs/advanced_log_collection.md [hunk 9]
```diff
@@ -58,13 +60,78 @@ logs:
...
+
+### include_at_match
 
 | Parameter          | Description                                                                       |
 |--------------------|-----------------------------------------------------------------------------------|
 | `include_at_match` | Only logs with a message that includes the specified pattern are sent to Datadog. |
 
-For example, sending only logs that contain a Datadog email address:
+
+For example, **filtering IN** logs that contain a Datadog email address, use the following `log_processing_rules`:
```
Comment (cswatt) [inline content/en/agent/logs/advanced_log_collection.md:131]:
```suggestion
For example, to **filter IN** logs that contain a Datadog email address, use the following `log_processing_rules`:
```

File: content/en/agent/logs/advanced_log_collection.md [hunk 10]
```diff
@@ -104,124 +233,76 @@ logs:
...
-```yaml
-- type: mask_sequences
-  name: on_Example_redaction
-  replace_placeholder: "on Example [Example REDACTED]"
-  pattern: (?:on Example\s?[^\s]+)
-```
+With Agent version 7.17+, the `replace_placeholder` string can expand references to capture groups such as `$1`, `$2` and so forth. If you want a string to follow the capture group with no space in between, use the format `${<GROUP_NUMBER>}`.
 
-##### Class A IP Address
+For instance, to scrub user information from the log `User email: foo.bar@example.com`, using:
```
Comment (cswatt) [inline content/en/agent/logs/advanced_log_collection.md:300]:
```suggestion
For instance, to scrub user information from the log `User email: foo.bar@example.com`, use:
```

File: content/en/agent/logs/advanced_log_collection.md [hunk 11]
```diff
@@ -104,124 +233,76 @@ logs:
...
+* `pattern: "(User email: )[^@]*@(.*)"`
+* `replace_placeholder: "$1 masked_user@${2}"`
 
-```yaml
-- type: mask_sequences
-  name: simple_ip_address
-  replace_placeholder: "[IP REDACTED]"
-  pattern: (?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)
-```
+Would send the following log to Datadog: `User email: masked_user@example.com`
```
Comment (cswatt) [inline content/en/agent/logs/advanced_log_collection.md:305]:
```suggestion
This sends the following log to Datadog: `User email: masked_user@example.com`
```

**Pull request merged**